roles/libvirt_manager/tasks/create_cloud_init_iso.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Generate cloud-init iso
+  become: false
+  vars:
+    _images_dir: "{{ [ cifmw_libvirt_manager_basedir, 'workload', ] | ansible.builtin.path_join }}"
+  block:
+    - name: "Acquire authorized_keys from hypervisor"
+      ansible.builtin.slurp:
+        src: "{{ ansible_user_dir ~ '/.ssh/authorized_keys' }}"
+      register: _authorized_keys_b64
+
+    - name: Create a list of ssh pub keys from authorized_keys
+      ansible.builtin.set_fact:
+        _ssh_authorized_keys: |-
+          [
+          {% for ssh_key in (_authorized_keys_b64.content | b64decode).split('\n') | select() %}
+            "{{ ssh_key | trim }}",
+          {% endfor %}
+          ]
+
+    - name: "Generate instance random UUID"
+      ansible.builtin.set_fact:
+        _instance_uuid: "{{ 99999999 | random(seed=vm) | to_uuid | lower }}"
+
+    - name: "Define the network config if needed"
+      when: item.value.ip_address is defined
+      ansible.builtin.set_fact:
+        _network_config:
+          network:
+            version: 2
+            renderer: NetworkManager
+            ethernets:
+              match:
+                name: eth0
+              set-name: eth0
+              addresses:
+                - "{{ item.value.ip_address.address }}"
+              gateway4: "{{ item.value.ip_address.gw }}"
+              nameservers:
+                addresses:
+                  - "{{ item.value.ip_address.dns }}"
+
+    - name: "Define the user data"
+      ansible.builtin.set_fact:
+        _user_data:
+          users:
```

Comment:
It would be nice to implement this so that it can be overriden? In case we ever need a different set of users? To make it "easy" we could require the entire list when overriding.

Reply:
ack, sounds good!

Reply:
@hjensas so... now we have 3 choices:
- accept the defaults with `root` and `zuul` users
- define a `user` field in the libvirt layout to change the "sudo" enabled user from `zuul` to something else
- define a `cifmw_libvirt_manager_users` to completely override the list of users.